src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -19,28 +20,60 @@
 
     private String mergeRemote;
     private String mergeTarget;
-    private String mergeStrategy;
+    private MergeCommand.Strategy mergeStrategy;
     private MergeCommand.GitPluginFastForwardMode fastForwardMode;
 
     /**
      * @deprecated use the new constructor that allows to set the fast forward mode.
      */
     @Deprecated
     public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(MergeCommand.GitPluginFastForwardMode.FF);
+    }
+
+    /**
+     * @Deprecated use the new @DataBoundConstructor where you only need to supply the necessary information.
```

Comment:
Yes, I remember. I had to move the `@DataBoundConstructor` to a new constructor to fix the bug. I marked the old constructor with `@Deprecated` but I did not remove it since it is still called by [UserMergeOptionsTest.java](https://github.com/jenkinsci/git-plugin/blob/master/src/test/java/hudson/plugins/git/UserMergeOptionsTest.java#L32) and I don't know what to do with that test.